crates/topos-p2p/src/behaviour/discovery.rs

```diff
@@ -105,9 +108,14 @@ impl DiscoveryBehaviour {
     }
 
     /// Change the interval of the next bootstrap queries
-    pub fn change_interval(&mut self, duration: Duration) -> Result<(), P2PError> {
+    pub async fn change_interval(&mut self, duration: Duration) -> Result<(), P2PError> {
         if let Some(interval) = self.next_bootstrap_query.as_mut() {
-            interval.set(tokio::time::interval(duration));
+            let mut new_interval = tokio::time::interval(duration);
+            // Delay the next tick
+            new_interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+            // ignore first tick
+            _ = new_interval.tick().await;
```

Comment:
In theory this isn't really needed right? We could also just let it tick and fire off the first query right away without pausing? (Not suggesting we do, but just to understand)

Reply:
If our first query is completing (failing) at `10s`, we change the interval and we expect the `Q2` to execute `5sec` later. Not directly as we want to let time to the network to change a bit (node going up in the delay).

```
                  0        10s   15s   20s   25s   30s
Expected Queries: |   Q1     | --- | Q2  | Q3  | Q4  | Q5  |
Actual Queries:   |   Q1     | Q2  | Q3  | Q4  | Q5  |
Mode:             | Normal   |  Fast
```

Reply:
Yes, I think I understand that part. But if the number of retries is decently large, or even infinite, then letting the first one fire right away wouldn't be a problem right?